src/billing/Members/Members.test.tsx

```diff
@@ -0,0 +1,278 @@
+        setDeletingUser={jest.fn()}
+      />
+    );
+
+    // Assert
+    const userTable = screen.getByRole('table');
+    expect(userTable).toHaveAccessibleName('users in billing project test-project');
+    const users = within(userTable).getAllByRole('row');
+    expect(users).toHaveLength(3); // 1 header row + 2 user rows
+    // users sort initially by email
+    expect(users[1]).toHaveTextContent(/user@test.email.orgUser/);
+    expect(users[2]).toHaveTextContent(/x_owner@test.email.orgOwnerThis user is the only Owner/);
+  });
+
+  it('supports sorting members', async () => {
+    // Arrange
+    const projectUsers = [
+      { email: 'x_owner@test.email.org', roles: ['Owner'] },
+      { email: 'user@test.email.org', roles: ['User'] },
+    ];
+    const user = userEvent.setup();
+
+    // Act
+    renderWithAppContexts(
+      <Members
+        billingProjectName='test-project'
+        isOwner
+        projectUsers={projectUsers}
+        setAddingUser={jest.fn()}
+        setEditingUser={jest.fn()}
+        setDeletingUser={jest.fn()}
+      />
+    );
+
+    // Assert
+    const roleButton = screen.getByText('Roles');
+    await user.click(roleButton);
+    const userTable = screen.getByRole('table');
+    const users = within(userTable).getAllByRole('row');
+    expect(users).toHaveLength(3); // 1 header row + 2 user rows
+    // users sort initially by email, clicking on role causes sorting by role
+    expect(users[1]).toHaveTextContent(/x_owner@test.email.orgOwner/);
+    expect(users[2]).toHaveTextContent(/user@test.email.orgUser/);
+  });
+
+  it('supports adding a member for owners', async () => {
+    // Arrange
+    const projectUsers = [{ email: 'owner@test.email.org', roles: ['Owner'] }];
+    const user = userEvent.setup();
+    const addUserCallback = jest.fn();
+
+    // Act
+    renderWithAppContexts(
+      <Members
+        billingProjectName='test-project'
+        isOwner
+        projectUsers={projectUsers}
+        setAddingUser={addUserCallback}
+        setEditingUser={jest.fn()}
+        setDeletingUser={jest.fn()}
+      />
+    );
+    const addUserButton = screen.getByText('Add User');
+    await user.click(addUserButton);
+
+    // Assert
+    expect(addUserCallback).toHaveBeenCalled();
+    // The actual display of the dialog to add a user is done in the parent file.
+  });
+
+  it('does not show the Add User button for non-owners', async () => {
+    // Arrange
+    const projectUsers = [{ email: 'owner@test.email.org', roles: ['Owner'] }];
+
+    // Act
+    renderWithAppContexts(
+      <Members
+        billingProjectName='test-project'
+        isOwner={false}
+        projectUsers={projectUsers}
+        setAddingUser={jest.fn()}
+        setEditingUser={jest.fn()}
+        setDeletingUser={jest.fn()}
+      />
+    );
+
+    // Assert
+    expect(screen.queryByText('Add User')).toBeNull();
+  });
+
+  it('disables the action menu for an owner if there are not multiple owners', async () => {
```

Comment:
The action menu testing cases be pushed down into a test for `MemberCard`, but the `group-common` code is still JS (with no tests), and I'm not inclined to convert it all right now. I would like to first get through `Project.js`.